pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/PinotDataType.java

```diff
@@ -263,21 +264,33 @@ public String convert(Object value, PinotDataType sourceType) {
   OBJECT {
     @Override
     public Integer toInteger(Object value) {
+      if (value instanceof Number) {
```

Comment:
Why do we need this? Number should not be categorized as OBJECT

Reply:
I've seen SHORT/INTEGER/LONGs are set as OBJECT from OrcRecordReader

Reply:
They should fall into the corresponding PinotDataType. Please check `DataTypeTransformer.transform()`. Also I think you need to add List into that method

Reply:
got it. will do